### PR TITLE
Add inference and accuracy evaluation for QDQ e2e example

### PR DESCRIPTION
**Pull request by chilo-ms**
Add inference and accuracy evaluation (exact match and f1 metrics) for QDQ BERT squad model.

### CODE REVIEW

**Comment by stevenlix** [comment]
There is a newer version evaluate-v2.0.py. Is it better to use it?

File: quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py [hunk 1]
```diff
@@ -63,11 +65,48 @@ def get_next(self):
                 input_ids = np.expand_dims(features[0].input_ids, axis=0)
                 input_mask = np.expand_dims(features[0].input_mask, axis=0)
                 segment_ids = np.expand_dims(features[0].segment_ids, axis=0)
-        data = [{"input_ids": input_ids, "input_mask": input_mask, "segment_ids":segment_ids}]
+        data = [{"input_ids": input_ids, "attention_mask": input_mask, "token_type_ids":segment_ids}]
         self.current_index += self.batch_size
         self.enum_data_dicts = iter(data)
         return next(self.enum_data_dicts, None)
 
+def inference(ort_session, features, example):
```
Comment (stevenlix) [inline quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py:73]: Can we make inference and evaluation more structural? for example, reuse BertDataReader to read data and define BertEvaluator for evaluation, just like what we did in other e2e examples.

Comment (chilo-ms) [reply quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py:73]: The reason I didn't use BertDataReader for inference and evaluation is that for some SQUAD examples, they might have more than one feature. Our current BertDataReader only considers one feature even though the example has multiple features. If we want to handle all the features for all the examples and at the same time run it in batch, we might need to modify the BertDataReader.

**Comment by chilo-ms** [comment]
> There is a newer version evaluate-v2.0.py. Is it better to use it?

I can try to add evaluate-v2.0.py

File: quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py (hunk 1 above)
Comment (stevenlix) [reply quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py:73]: It's good to handle all the features. One thing is we need to make sure calibration are using same amount of features as inference/evaluation, otherwise calibration may not be able to fully represent inference data

Comment (chilo-ms) [reply quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py:73]: For evaluation, I think we must handle all the features of each example in order to achieve better accuracy. Therefore, current data reader should be modified for calibration and evaluation to handle all the features.

Comment (stevenlix) [reply quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py]: Yes. we should not use same data for both calibration and evaluation. As we did in other e2e model, we should split the data set, i.e. 80% vs 20%, for each. But the key is to use same pre-processing and data/feature handling for both.

**Comment by chilo-ms** [comment]
> > There is a newer version evaluate-v2.0.py. Is it better to use it?
> 
> I can try to add evaluate-v2.0.py

evaluate-v2.0.py added as an option.

File: quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py [hunk 2]
```diff
@@ -125,4 +243,32 @@ def get_next(self):
...
+    sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL
+    session = onnxruntime.InferenceSession(qdq_model_path, sess_options=sess_options, providers=["TensorrtExecutionProvider", "CUDAExecutionProvider"])
+    all_predictions = inference(data_reader, session) 
+
+    prediction_file = "./prediction.json"
+    with open(prediction_file, "w") as f:
+        f.write(json.dumps(all_predictions, indent=4))
+        print("\nOutput dump to {}".format(prediction_file))
+
+    # Evaluate QDQ model for SQUAD v1.1
```
Comment (stevenlix) [inline quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py:263]: why do we run evaluation in the separate script? Is it better to show evaluation result right after the inference is done since it's an e2e example?

Comment (chilo-ms) [reply quantization/nlp/bert/trt/e2e_tensorrt_bert_example.py:263]: Yes, we can show result after the inference is done.
Will modify to execute these evalute-XXX.py directly inside our e2e example script.

**Comment by drcege** [comment]
Hi, could you guys add the exact model ID or link to the used huggingface models? There are too many bert models to choose. I tried several but the output nodes did not match.
